src/blocks/data.js

```diff
@@ -768,6 +777,62 @@ class Scratch3DataBlocks {
 
         this.runtime.requestToolboxExtensionsUpdate();
     }
+
+    serializeVariable (block, target) {
+        const variableName = block.mutation.blockInfo.text;
```

Comment:
Probably this function should just take `blockInfo` as an argument, and then some helper function outside of the extension definition should handle high level block things like setting the x and y.

Reply:
The problem is that variables have their own custom way of serializing x and y, so it kind of needs to be here. That's also why the custom deserialize function needs to return the whole block instead of just the blockInfo guys :/